web/libs/editor/src/components/InteractiveOverlays/CommentsOverlay.tsx

```diff
@@ -152,10 +208,16 @@ const CommentsOverlayInner: React.FC<CommentsOverlayProps> = observer(({ annotat
   return (
     // biome-ignore lint/a11y/noSvgWithoutTitle: It's not just an icon or a figure; it's an entire interactive layer.
     <svg className={containerStyles.join(" ")} ref={setRef} xmlns="http://www.w3.org/2000/svg">
-      {overlayComments.map((comment: MSTComment) => {
-        const { id } = comment;
-        return <CommentItem key={[id, uniqKey].join("-")} comment={comment} rootRef={rootRef} />;
-      })}
+      <g key={uniqKey}>
+        {annotation.linkingMode === LINK_COMMENT_MODE &&
+          annotation.results.map((result: MSTResult) => (
```

Comment:
```suggestion
          annotation.results
          .filter((result: MSTResult)=>result.from_name.isClassificationTag)
          .map((result: MSTResult) => (
```
Most probably we can skip some results here.

Reply:
```suggestion
          annotation.results
          .filter((result: MSTResult)=> {
            return result.from_name.isClassificationTag && (!result.area || result.area.selected)
          })
          .map((result: MSTResult) => (
```
Or even like that.
It'll will help also with overlapping in per-regions.

Reply:
Oh, thank you! I did it in a good way, checking different main cases we have